tests/test_algorithm.py

```diff
@@ -1294,6 +1297,38 @@ def test_account_in_init(self):
         )
         test_algo.run(self.data_portal)
 
+    def test_good_kwargs(self):
+        """
+        Test that api methods on the data object can be called with keyword
+        arguments
+        """
+        test_algo = TradingAlgorithm(
+            script=call_with_kwargs,
+            sim_params=self.sim_params,
+            env=self.env,
+        )
+        test_algo.run(self.data_portal)
+
+    def test_bad_kwargs(self):
+        """
+        Test that api methods on the data object called with bad kwargs return
+        a meaningful TypeError that we create, rather than an unhelpful cython
+        error
+        """
+        for script in [call_with_bad_kwargs_history,
+                       call_with_bad_kwargs_current]:
+
+            with self.assertRaises(TypeError) as cm:
+                test_algo = TradingAlgorithm(
+                    script=script,
+                    sim_params=self.sim_params,
+                    env=self.env,
+                )
+                test_algo.run(self.data_portal)
+
+            self.assertIn("got an unexpected keyword argument 'blahblah'",
+                          cm.exception.message)
```

Comment:
`BaseException.message` is deprecated in Python 3.  You want `cm.exception.args[0]` here.